core/services/directrequest/delegate.go

```diff
@@ -297,59 +325,48 @@ func (l *listener) handleOracleRequest(request *operator_wrapper.OperatorOracleR
 				"minContractPayment", minContractPayment.String(),
 				"requestPayment", requestPayment.String(),
 			)
-			ctx, cancel := postgres.DefaultQueryCtx()
-			defer cancel()
-			if err := l.logBroadcaster.MarkConsumed(l.db.WithContext(ctx), lb); err != nil {
-				l.logger.Errorw("DirectRequest: unable to mark log consumed", "err", err, "log", lb.String())
-			}
+			l.markLogConsumed(nil, lb)
 			return
 		}
 	}
 
 	meta := make(map[string]interface{})
 	meta["oracleRequest"] = oracleRequestToMap(request)
 
-	l.shutdownWaitGroup.Add(1)
-	go func() {
```

Comment:
Hm we put this in a goroutine because the pipeline running is blocking and these DR jobs have API calls in them. The VRF can get away with inlining the calls because its all in memory. I think we still want this in a goroutine, I think with a single log processor goroutine and the shared l.runs which will cancel an in flight run if its running we should be good

Reply:
so, the goroutine is problematic from the perspective of 'MarkConsumed', like i mentioned in the other comment.
But actually this could be solved by keeping track of running jobs in a map keyed by block hash + log index. And when another duplicated log appears, processing of it (and subsequent ones) could be blocked until the first one's is finished.

Reply:
I was thinking we just put the WasConsumed check inside this goroutine - is that feasible?

Reply:
Spoke offline, problem remains. Will leave as sequential with the idea that we can slowly work through a large mailbox of logs